src/sst/elements/vanadis/inst/vmovci.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2009-2021 NTESS. Under the terms
+// of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Copyright (c) 2009-2021, NTESS
+// All rights reserved.
+//
+// Portions are copyright of other developers:
+// See the file CONTRIBUTORS.TXT in the top level directory
+// the distribution for more information.
+//
+// This file is part of the SST software package. For license
+// information, see the LICENSE file in the top level directory of the
+// distribution.
+
+#ifndef _H_VANADIS_MOVI
+#define _H_VANADIS_MOVI
+
+#include "inst/vinst.h"
+
+namespace SST {
+namespace Vanadis {
+
+class VanadisMoveCompareImmInstruction : public VanadisInstruction {
+public:
+    VanadisMoveCompareImmInstruction(const uint64_t addr, const uint32_t hw_thr, const VanadisDecoderOptions* isa_opts,
+                            const uint16_t dest, const uint16_t src, const uint16_t reg_comp, const uint64_t immediate,
+			    VanadisRegisterCompareType c_type)
+        : VanadisInstruction(addr, hw_thr, isa_opts, 2, 1, 2, 1, 0, 0, 0, 0),
+		imm_value(immediate), cmp_type(c_type) {
+
+        isa_int_regs_in[0] = src;
+	isa_int_regs_in[1] = reg_comp;
+        isa_int_regs_out[0] = dest;
+    }
+
+    VanadisMoveCompareImmInstruction* clone() { return new VanadisMoveCompareImmInstruction(*this); }
+
+    virtual VanadisFunctionalUnitType getInstFuncType() const { return INST_INT_ARITH; }
```

Comment:
Are these virtual methods the ones inherited from `VanadisInstruction` if so prefer `override` to `virtual`